scan/src/pages/BlockHomePage.re

```diff
@@ -69,13 +69,27 @@ let renderBody = ((block, proposer): (BlockHook.Block.t, option(ValidatorHook.Va
 
 [@react.component]
 let make = () => {
-  let (limit, setLimit) = React.useState(_ => 10);
-  let blocksOpt = BlockHook.latest(~limit, ());
+  let (page, setPage) = React.useState(_ => 1);
+  let limit = 10;
+
+  let blocksOpt = BlockHook.latest(~limit, ~page, ());
   let infoOpt = React.useContext(GlobalContext.context);
 
+  let blocksCountOpt = BlockHook.latest();
+  let latestHeightOpt = {
+    let%Opt blocks = blocksCountOpt;
+    let%Opt latestBlock = blocks->Belt.List.get(0);
+    Some(latestBlock.height);
+  };
+  let pageCount =
+    {
+      let%Opt latestHeight = latestHeightOpt;
+      Some(Page.getPageCount(latestHeight, limit));
+    }
+    |> Belt.Option.getWithDefault(_, 1);
```

Comment:
We will need to revisit this sometime after we clearly define how to deal with loading state.

Reply:
Sure